File: app/src/main/java/com/example/fragment/MainActivity.java
package com.example.fragment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button btnFragA, btnFragB, btnFragC;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnFragA = findViewById(R.id.fragA);
        btnFragB = findViewById(R.id.fragB);
        btnFragC = findViewById(R.id.fragC);

        //Default flag open
        loadFrag(new BFragment(),1);

        btnFragA.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loadFrag(new AFragment(),1);
            }
        });

        btnFragB.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loadFrag(new BFragment(),1);
            }
        });

        btnFragC.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loadFrag(new CFragment(),1);
            }
        });

    }

    public void loadFrag(Fragment frag, int flag){
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();

        if(flag==0) {
            ft.add(R.id.container, frag);
        }
        else{
            ft.replace(R.id.container,frag);
        }
        ft.commit();
    }
}

File: app/src/main/java/com/example/fragment/MainActivity.kt
package com.example.fragment

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //Default flag open
        loadFrag(BFragment(), 0)
        val btnFragA = findViewById<Button>(R.id.fragA).setOnClickListener {
            loadFrag(
                AFragment(),
                1
            )
        }
        val btnFragB = findViewById<Button>(R.id.fragB).setOnClickListener {
            loadFrag(
                BFragment(),
                1
            )
        }
        val btnFragC = findViewById<Button>(R.id.fragC).setOnClickListener {
            loadFrag(
                CFragment(),
                1
            )
        }
    }

    private fun loadFrag(frag: Fragment?, flag: Int) {
        val fm = supportFragmentManager
        val ft = fm.beginTransaction()
        if (flag == 0) {
            ft.add(R.id.container, frag!!)
        } else {
            ft.replace(R.id.container, frag!!)
        }
        ft.commit()
    }
}